spec.html

```diff
@@ -25659,29 +25659,28 @@ <h1>Runtime Semantics: Decode ( _string_, _reservedSet_ )</h1>
                 1. If the code units at index (_k_ + 1) and (_k_ + 2) within _string_ do not represent hexadecimal digits, throw a *URIError* exception.
                 1. Let _B_ be the 8-bit value represented by the two hexadecimal digits at index (_k_ + 1) and (_k_ + 2).
                 1. Set _k_ to _k_ + 2.
-                1. If the most significant bit in _B_ is 0, then
+                1. Let _n_ be the number of leading 1 bits in _B_.
+                1. If _n_ = 0, then
                   1. Let _C_ be the code unit whose value is _B_.
                   1. If _C_ is not in _reservedSet_, then
                     1. Let _S_ be the String value containing only the code unit _C_.
                   1. Else,
                     1. Let _S_ be the substring of _string_ from _start_ to _k_ + 1.
                 1. Else,
-                  1. Assert: the most significant bit in _B_ is 1.
-                  1. Let _n_ be the smallest nonnegative integer such that (_B_ &lt;&lt; _n_) &amp; 0x80 is equal to 0.
-                  1. If _n_ equals 1 or _n_ is greater than 4, throw a *URIError* exception.
-                  1. Let _Octets_ be a List of 8-bit integers of size _n_.
-                  1. Set _Octets_[0] to _B_.
+                  1. If _n_ = 1 or _n_ &gt; 4, throw a *URIError* exception.
                   1. If _k_ + (3 &times; (_n_ - 1)) is greater than or equal to _strLen_, throw a *URIError* exception.
+                  1. Let _Octets_ be a new List containing _B_.
                   1. Let _j_ be 1.
                   1. Repeat, while _j_ &lt; _n_,
                     1. Set _k_ to _k_ + 1.
                     1. If the code unit at index _k_ within _string_ is not the code unit 0x0025 (PERCENT SIGN), throw a *URIError* exception.
                     1. If the code units at index (_k_ + 1) and (_k_ + 2) within _string_ do not represent hexadecimal digits, throw a *URIError* exception.
                     1. Let _B_ be the 8-bit value represented by the two hexadecimal digits at index (_k_ + 1) and (_k_ + 2).
-                    1. If the two most significant bits in _B_ are not 10, throw a *URIError* exception.
+                    1. If the two most significant bits of _B_ in descending significance are not 1 and 0 respectively, throw a *URIError* exception.
```

Comment:
```suggestion
                    1. If the two most significant bits of _B_ in order of descending significance are not 1 and 0 respectively, throw a *URIError* exception.
```

Reply:
I see @michaelficarra suggested the current wording; I'll try to resolve this definitively at today's editor call.

Reply:
WFM

Reply:
i lightly prefer @bakkot's suggestion

Reply:
Accepted. It also occurs to me that this step could be removed completely, because there's a later check that throws a URIError if the final value of _Octets_ "does not contain a valid UTF-8 encoding of a Unicode code point", which is just as true for **« 0xC0, 0x00 »** as it is for **« 0xC0, 0x80 »**.

Reply:
@gibson042 sure, that sounds even better.

Reply:
Done.